sound/soc/sof/intel/hda-pcm.c

```diff
@@ -240,6 +245,14 @@ int hda_dsp_pcm_open(struct snd_sof_dev *sdev,
 	if (hda_always_enable_dmi_l1 && direction == SNDRV_PCM_STREAM_CAPTURE)
 		runtime->hw.info &= ~SNDRV_PCM_INFO_PAUSE;
 
+	/*
+	 * Follow the pause_supported flag (unless a module parameter is set to
+	 * ignore it) and disable the pause push/release support
+	 */
+	if (!hda_force_pause_support &&
+	    !spcm->stream[substream->stream].pause_supported)
+		runtime->hw.info &= ~SNDRV_PCM_INFO_PAUSE;
```

Comment:
it's a bit weird because we could have a configuration with hda_always_enable_dmi_l1=1 and hda_force_pause_support=1, and in this case the force_pause_support does not work..

Reply:
in case of forced  dmi_l1 we cannot enable the pause support as it would lead to a hardware issue.
Currently the hda_always_enable_dmi_l1 would "silently" disable the pause support for only capture.

I'm not sure how is better to handle this...

Reply:
maybe add a condition to check that hda_always_enable_dmi_l1 and hda_force_pause_support have compatible values.

Reply:
@plbossart, if they are not compatible? I think we should always obey the `hda_always_enable_dmi_l1` and disable the pause support for capture regardless of what `hda_force_pause_support` commands, like I have done it.